src/js/events.js

```diff
@@ -146,6 +146,10 @@ define(function (require, exports, module) {
             LIBRARY_CREATED: "libraryCreated",
             LIBRARY_REMOVED: "libraryRemoved",
             LIBRARY_SELECTED: "librarySelected"
+        },
+        search: {
+            LAYER_SEARCH_OPTIONS: "layerSearchOptions",
```

Comment:
There should just be one new event here. Something like `REGISTER_SEARCH_PROVIDER`. All the providers should emit the same event because they'll all be handled uniformly by the search store.